clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -12,552 +12,142 @@
-      .loadNamedProfile(profileName);
-    const downloadBinary = !!SettingsService.getCopybookFileEncoding();
-    const filePath = Path.join(
-      CopybookURI.createDatasetPath(profileName, dataset, this.storagePath),
-      copybook.substring(
-        0,
-        copybook.indexOf(".") !== -1 ? copybook.indexOf(".") : copybook.length,
-      ),
-    );
-
-    const downloadOptions = {
-      file: filePath,
-      binary: downloadBinary,
-      returnEtag: true,
-    };
-    if (!SettingsService.getCopybookFileEncoding()) {
-      (downloadOptions as any).encoding = loadedProfile.profile.encoding;
-    }
-
-    if (isUSS) {
-      await zoweExplorerApi
-        .getUssApi(loadedProfile)
-        .getContents(`${dataset}/${copybook}`, downloadOptions);
-    } else {
-      await zoweExplorerApi
-        .getMvsApi(loadedProfile)
-        .getContents(`${dataset}(${copybook})`, downloadOptions);
-    }
-
-    if (downloadBinary) {
-      let newContent = iconv.decode(
-        fs.readFileSync(filePath),
-        SettingsService.getCopybookFileEncoding() as string,
+  static async getInstance(
+    storagePath: string,
+  ): Promise<CopybookDownloadService> {
+    if (this._instance === null) {
+      this._instance = new CopybookDownloadService(
+        storagePath,
+        await Utils.getZoweExplorerAPI(),
       );
-      if (!isUSS) {
-        // Based on assumption - Most of source code on z/OS is 80 characters per record - JCL, HLASM, COBOL
-        // Can be exposed later on as a setting.
-        newContent = newContent.replace(/.{80}/g, `$&\n`);
-      }
-      fs.writeFileSync(filePath, newContent);
     }
+    return this._instance;
   }
 
-  public async handleCopybooks(
-    toDownload: CopybookProfile[],
-    errors: Set<string>,
-    progress: vscode.Progress<{ message?: string; increment?: number }>,
-    isUSS: boolean,
-  ) {
-    const promises: Promise<void>[] = [];
-    try {
-      for (const cp of toDownload) {
-        const datasets = isUSS
-          ? SettingsService.getUssPath(cp.documentUri, cp.dialectType)
-          : SettingsService.getDsnPath(cp.documentUri, cp.dialectType);
-        for (const dataset of datasets) {
-          if (CopybookDownloadService.needsUserNotification(toDownload)) {
-            progress.report({
-              message:
-                "Looking in " +
-                dataset +
-                ". " +
-                toDownload.length +
-                " copybook(s) left.",
-            });
-          }
-          promises.push(
-            this.handleCopybook(progress, dataset, cp, errors, isUSS),
-          );
-        }
-      }
-      await Promise.all(promises);
-    } catch (e: any) {
-      if (CopybookDownloadService.isInvalidCredentials(e)) {
-        throw e;
-      }
-      if (CopybookDownloadService.needsUserNotification(toDownload)) {
-        vscode.window.showErrorMessage(e.toString());
-      }
-    }
+  dispose() {
+    this.explorerAPI = undefined;
```

Comment:
I was literally just asking... If the `IApiRegisterClient` does not have any dispose or similar mechanism, it is ok not to have dispose method.

Reply:
Ok. Just checked the ZE exported api, there is no dispose. Also updated